src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
-    }
+        // as the statement of the opening proof is already bound to the interal state of the fs_rng,
+        // we simply squeeze the challenge scalar for the random linear combination
+        let lambda: G::ScalarField = fs_rng.squeeze_128_bits_challenge();
+        let mut cur_challenge = G::ScalarField::one();
 
-    #[derive(Copy, Clone, Default)]
-    struct TestInfo {
-        num_iters: usize,
-        /// Max segment size
-        max_degree: Option<usize>,
-        /// Segment size
-        supported_degree: Option<usize>,
-        num_polynomials: usize,
-        enforce_degree_bounds: bool,
-        max_num_queries: usize,
-        num_equations: Option<usize>,
-        segmented: bool,
-        negative_type: Option<NegativeType>,
-    }
+        let mut has_hiding = false;
 
-    pub fn bad_degree_bound_test<F, PC>() -> Result<(), PC::Error>
-    where
-        F: Field,
-        PC: PolynomialCommitment<F>,
-    {
-        let rng = &mut thread_rng();
-        let max_degree = 100;
-        let pp = PC::setup(max_degree)?;
-
-        test_canonical_serialize_deserialize(true, &pp);
-
-        for _ in 0..10 {
-            let supported_degree = rand::distributions::Uniform::from(1..=max_degree).sample(rng);
-            assert!(
-                max_degree >= supported_degree,
-                "max_degree < supported_degree"
-            );
-
-            let mut labels = Vec::new();
-            let mut polynomials = Vec::new();
-            let mut degree_bounds = Vec::new();
-
-            for i in 0..10 {
-                let label = format!("Test{}", i);
-                labels.push(label.clone());
-                let poly = Polynomial::rand(supported_degree, rng);
-
-                let degree_bound = 1usize;
-                let hiding_bound = Some(1);
-                degree_bounds.push(degree_bound);
-
-                polynomials.push(LabeledPolynomial::new(
-                    label,
-                    poly,
-                    Some(degree_bound),
-                    hiding_bound,
-                ))
+        let h_poly_time = start_timer!(|| "Compute h(X) polynomial");
+
+        // h(X)
+        let mut h_polynomial = Polynomial::zero();
+
+        // z(X), the vanishing polynomial of the query set.
+        let mut z_polynomial = Polynomial::from_coefficients_slice(&[G::ScalarField::one()]);
+
+        // Save evaluation points for later
+        let mut eval_points = std::collections::HashSet::new();
+
+        // Collect middleware results into vectors
+        let mut x_polynomials = vec![];
+        let mut polynomials = vec![];
+        let mut randomnesses = vec![];
+
+        for (label, (_point_label, point)) in query_set.iter() {
+            eval_points.insert(*point);
+
+            if !commitment_map.contains_key(label) {
+                end_timer!(combine_time);
+                Err(Error::MissingCommitment {
+                    label: label.to_string(),
+                })?
```

Comment:
Do I understand correctly, we only provide the commitments to our combine function to check that for every polynomial there exists a commitment?

Reply:
Removed